### PR TITLE
Add JS Map/Set/WeakMap/WeakSet compat data

### PR DESCRIPTION
**Pull request by Elchi3**
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet

### CODE REVIEW

File: javascript/builtins/Map.json [hunk 1]
```diff
@@ -0,0 +1,1060 @@
...
+    "builtins": {
+      "Map": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Map",
+          "support": {
+            "webview_android": {
+              "version_added": "38"
+            },
+            "chrome": {
+              "version_added": "38"
```
Comment (teoli2003) [inline javascript/builtins/Map.json]: So we don't mark the experimental Chrome flags?

File: javascript/builtins/Map.json [hunk 2]
```diff
@@ -0,0 +1,1060 @@
...
+                "version_added": "12"
+              },
+              "firefox": {
+                "version_added": "13"
+              },
+              "firefox_android": {
+                "version_added": "13"
+              },
+              "ie": {
+                "version_added": "11"
```
Comment (teoli2003) [inline javascript/builtins/Map.json]: The original table had 'No support' here. On purpose, or bad copy and paste?

File: javascript/builtins/Map.json [hunk 3]
```diff
@@ -0,0 +1,1060 @@
...
+              },
+              "ie_mobile": {
+                "version_added": "11"
```
Comment (teoli2003) [inline javascript/builtins/Map.json]: The original table had 'No support' here. On purpose, or bad copy and paste?

File: javascript/builtins/Map.json [hunk 4]
```diff
@@ -0,0 +1,1060 @@
...
+              },
+              "nodejs": {
+                "version_added": true
+              },
+              "opera": {
+                "version_added": "25"
```
Comment (teoli2003) [inline javascript/builtins/Map.json]: I think this should be 'true' instead (if we know the opera version, we know the chrome one)

File: javascript/builtins/Map.json [hunk 5]
```diff
@@ -0,0 +1,1060 @@
...
+              },
+              "opera_android": {
+                "version_added": "25"
```
Comment (teoli2003) [inline javascript/builtins/Map.json]: I think this should be 'true' instead (if we know the opera version, we know the chrome one)

File: javascript/builtins/Map.json [hunk 6]
```diff
@@ -0,0 +1,1060 @@
...
+                "version_added": "38"
+              },
+              "edge": {
+                "version_added": "12"
+              },
+              "edge_mobile": {
+                "version_added": "12"
+              },
+              "firefox": {
+                "version_added": "19"
```
Comment (teoli2003) [inline javascript/builtins/Map.json]: Adding this as a comment? 
"From Gecko 13 (Firefox 13 / Thunderbird 13 / SeaMonkey 2.10) to Gecko 18 (Firefox 18 / Thunderbird 18 / SeaMonkey 2.15 / Firefox OS 1.0.1 / Firefox OS 1.1) the size property was implemented as a Map.prototype.size() method, this has been changed to a property in later versions conform to the ECMAScript 2015 specification (bug 807001)."

File: javascript/builtins/Map.json [hunk 7]
```diff
@@ -0,0 +1,1060 @@
...
+              },
+              "firefox_android": {
+                "version_added": "19"
```
Comment (teoli2003) [inline javascript/builtins/Map.json]: Adding this as a comment? 
"From Gecko 13 (Firefox 13 / Thunderbird 13 / SeaMonkey 2.10) to Gecko 18 (Firefox 18 / Thunderbird 18 / SeaMonkey 2.15 / Firefox OS 1.0.1 / Firefox OS 1.1) the size property was implemented as a Map.prototype.size() method, this has been changed to a property in later versions conform to the ECMAScript 2015 specification (bug 807001)."

File: javascript/builtins/Set.json [hunk 8]
```diff
@@ -0,0 +1,898 @@
...
+    "builtins": {
+      "Set": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Set",
+          "support": {
+            "webview_android": {
+              "version_added": "38"
+            },
+            "chrome": {
+              "version_added": "38"
```
Comment (teoli2003) [inline javascript/builtins/Set.json]: Same question about info of support behind a flag.

File: javascript/builtins/Set.json [hunk 9]
```diff
@@ -0,0 +1,898 @@
...
+            },
+            "chrome_android": {
+              "version_added": "38"
```
Comment (teoli2003) [inline javascript/builtins/Set.json]: Same question about info of support behind a flag.

File: javascript/builtins/Set.json [hunk 10]
```diff
@@ -0,0 +1,898 @@
...
+                "version_added": "12"
+              },
+              "firefox": {
+                "version_added": "13"
+              },
+              "firefox_android": {
+                "version_added": "13"
+              },
+              "ie": {
+                "version_added": "11"
```
Comment (teoli2003) [inline javascript/builtins/Set.json]: Original table had 'false'

File: javascript/builtins/Set.json [hunk 11]
```diff
@@ -0,0 +1,898 @@
...
+              },
+              "ie_mobile": {
+                "version_added": "11"
```
Comment (teoli2003) [inline javascript/builtins/Set.json]: Original table had 'false'

File: javascript/builtins/Set.json [hunk 12]
```diff
@@ -0,0 +1,898 @@
...
+              },
+              "nodejs": {
+                "version_added": true
+              },
+              "opera": {
+                "version_added": "25"
```
Comment (teoli2003) [inline javascript/builtins/Set.json]: Shouldn't this be 'true'?

File: javascript/builtins/Set.json [hunk 13]
```diff
@@ -0,0 +1,898 @@
...
+              },
+              "opera_android": {
+                "version_added": "25"
```
Comment (teoli2003) [inline javascript/builtins/Set.json]: Shouldn't this be 'true'?

File: javascript/builtins/Set.json [hunk 14]
```diff
@@ -0,0 +1,898 @@
...
+                "version_added": "38"
+              },
+              "edge": {
+                "version_added": "12"
+              },
+              "edge_mobile": {
+                "version_added": "12"
+              },
+              "firefox": {
+                "version_added": "19"
```
Comment (teoli2003) [inline javascript/builtins/Set.json]: Add comment?
"[1] From Gecko 13 (Firefox 13 / Thunderbird 13 / SeaMonkey 2.10) to Gecko 18 (Firefox 18 / Thunderbird 18 / SeaMonkey 2.15 / Firefox OS 1.0.1 / Firefox OS 1.1) the size property was implemented as a Set.prototype.size() method, this has been changed to a property in later versions conform to the ECMAScript 6 specification (bug 807001)."

File: javascript/builtins/Set.json [hunk 15]
```diff
@@ -0,0 +1,898 @@
...
+              },
+              "firefox_android": {
+                "version_added": "19"
```
Comment (teoli2003) [inline javascript/builtins/Set.json]: Add comment?
"[1] From Gecko 13 (Firefox 13 / Thunderbird 13 / SeaMonkey 2.10) to Gecko 18 (Firefox 18 / Thunderbird 18 / SeaMonkey 2.15 / Firefox OS 1.0.1 / Firefox OS 1.1) the size property was implemented as a Set.prototype.size() method, this has been changed to a property in later versions conform to the ECMAScript 6 specification (bug 807001)."

File: javascript/builtins/WeakMap.json [hunk 16]
```diff
@@ -0,0 +1,559 @@
...
+              },
+              "edge": {
+                "version_added": false
+              },
+              "edge_mobile": {
+                "version_added": false
+              },
+              "firefox": {
+                "version_added": "20",
+                "version_removed": "45"
```
Comment (teoli2003) [inline javascript/builtins/WeakMap.json]: The table has '46' which one is correct?

File: javascript/builtins/WeakMap.json [hunk 17]
```diff
@@ -0,0 +1,559 @@
...
+              },
+              "firefox_android": {
+                "version_added": "20",
+                "version_removed": "45"
```
Comment (teoli2003) [inline javascript/builtins/WeakMap.json]: The table has '46' which one is correct?

File: javascript/builtins/WeakMap.json [hunk 18]
```diff
@@ -0,0 +1,559 @@
...
+                "version_added": "11"
+              },
+              "ie_mobile": {
+                "version_added": "11"
+              },
+              "nodejs": {
+                "version_added": true
+              },
+              "opera": {
+                "version_added": "25"
```
Comment (teoli2003) [inline javascript/builtins/WeakMap.json]: "version_removed": "30" ?

File: javascript/builtins/WeakMap.json [hunk 19]
```diff
@@ -0,0 +1,559 @@
...
+              },
+              "opera_android": {
+                "version_added": "25"
```
Comment (teoli2003) [inline javascript/builtins/WeakMap.json]: "version_removed": "30" ?

File: javascript/builtins/WeakMap.json [hunk 20]
```diff
@@ -0,0 +1,559 @@
...
+              },
+              "safari": {
+                "version_added": "7.1"
```
Comment (teoli2003) [inline javascript/builtins/WeakMap.json]: "version_removed": "9" ?

File: javascript/builtins/WeakMap.json [hunk 21]
```diff
@@ -0,0 +1,559 @@
...
+              },
+              "safari_ios": {
+                "version_added": "8"
```
Comment (teoli2003) [inline javascript/builtins/WeakMap.json]: "version_removed": "9" ?

File: javascript/builtins/WeakSet.json [hunk 22]
```diff
@@ -0,0 +1,391 @@
...
+              "version_added": "36"
+            },
+            "edge": {
+              "version_added": "12"
+            },
+            "edge_mobile": {
+              "version_added": "12"
+            },
+            "firefox": {
+              "version_added": "36"
```
Comment (teoli2003) [inline javascript/builtins/WeakSet.json]: The table has '34'.

File: javascript/builtins/WeakSet.json [hunk 23]
```diff
@@ -0,0 +1,391 @@
...
+            },
+            "firefox_android": {
+              "version_added": "36"
```
Comment (teoli2003) [inline javascript/builtins/WeakSet.json]: The table has '34'.

File: javascript/builtins/Map.json (hunk 1 above)
Comment (Elchi3) [reply javascript/builtins/Map.json]: I wasn't sure if it is really worth it. Chrome 31 is quite old now and it adds complexity to the table. Happy to add it, if you think preserving this info is important, though.

**Comment by Elchi3** [comment]
Updated and rebased (for stricter validation testing)

File: javascript/builtins/Map.json (hunk 1 above)
Comment (teoli2003) [reply javascript/builtins/Map.json]: Fine without them for me.

**Pull request merged**